### ISSUE TITLE
How to Confirm the results?

### ISSUE THREAD

**Issue by ChenMuDX** [AntixK/PyTorch-VAE#104]
Hello  , I have trained a VanillaVAE model using your code. The reconstructions and samples look good. 
But how can I compare the results between yours(in theREADME.md) and mine? Could you provide the testing data? Thanks！

**Comment by RobertoGrabovac** [comment]
Hey, how did you manage to train VanillaVAE model? Did you use google colab and can you write exact versions of all libraries used? I would be grateful for this as I didn't manage to even start the training process due to the errors.

**Comment by ChenMuDX** [comment]
I didn't use the Colab library。
This is my requirement.txt pytorch-lightning==1.9.0
PyYAML==6.0
tensorboard>=2.2.0
torch>=1.6.1
torchsummary==1.5.1
torchvision>=0.10.1


and all libraries:


absl-py                  2.1.0
aiohappyeyeballs         2.4.4
aiohttp                  3.11.9
aiosignal                1.3.1
anyio                    4.6.2.post1
arrow                    1.3.0
asttokens                3.0.0
async-timeout            5.0.1
attrs                    24.2.0
beautifulsoup4           4.12.3
blessed                  1.20.0
boto3                    1.35.72
botocore                 1.35.72
certifi                  2024.6.2
charset-normalizer       3.3.2
click                    8.1.7
cmake                    3.31.1
comm                     0.2.2
contextlib2              21.6.0
contourpy                1.3.0
croniter                 1.3.15
cycler                   0.12.1
dateutils                0.6.12
debugpy                  1.8.11
decorator                4.4.2
deepdiff                 7.0.1
distro                   1.7.0
editor                   1.6.6
et-xmlfile               1.1.0
exceptiongroup           1.2.2
executing                2.1.0
fastapi                  0.88.0
filelock                 3.15.3
fonttools                4.54.1
frozenlist               1.5.0
fsspec                   2023.12.2
future                   1.0.0
grpcio                   1.68.0
h11                      0.14.0
idna                     3.7
inquirer                 3.4.0
ipykernel                6.29.5
ipython                  8.30.0
itsdangerous             2.2.0
jedi                     0.19.2
Jinja2                   3.1.4
jmespath                 1.0.1
joblib                   1.4.2
jupyter_client           8.6.3
jupyter_core             5.7.2
kiwisolver               1.4.7
lightning-cloud          0.5.70
lightning-utilities      0.11.9
lit                      18.1.8
Markdown                 3.7
markdown-it-py           3.0.0
MarkupSafe               2.1.5
matplotlib               3.9.2
matplotlib-inline        0.1.7
mdurl                    0.1.2
ml_collections           0.1.1
mpmath                   1.3.0
multidict                6.1.0
nest-asyncio             1.6.0
networkx                 3.3
numpy                    1.26.4
nvidia-cublas-cu11       11.11.3.6
nvidia-cuda-cupti-cu11   11.8.87
nvidia-cuda-nvrtc-cu11   11.8.89
nvidia-cuda-runtime-cu11 11.8.89
nvidia-cudnn-cu11        8.7.0.84
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.3.0.86
nvidia-cusolver-cu11     11.4.1.48
nvidia-cusparse-cu11     11.7.5.86
nvidia-nccl-cu11         2.20.5
nvidia-nvtx-cu11         11.8.86
nvidia-pyindex           1.0.9
onnx                     1.17.0
opencv-python            4.10.0.84
openpyxl                 3.1.5
ordered-set              4.1.0
packaging                22.0
pandas                   2.2.3
parso                    0.8.4
pexpect                  4.9.0
pillow                   10.3.0
pip                      24.0
pip-autoremove           0.10.0
platformdirs             4.3.6
prompt_toolkit           3.0.48
propcache                0.2.1
protobuf                 5.29.0
psutil                   6.1.0
ptyprocess               0.7.0
pure_eval                0.2.3
pycairo                  1.20.1
pydantic                 1.10.19
pyDeprecate              0.3.1
Pygments                 2.18.0
PyGObject                3.42.1
PyJWT                    2.10.1
pyparsing                3.1.4
PySocks                  1.7.1
python-dateutil          2.9.0.post0
python-multipart         0.0.19
pytorch-lightning        1.9.0
pytz                     2024.2
PyYAML                   6.0
pyzmq                    26.2.0
readchar                 4.2.1
requests                 2.32.3
rich                     13.9.4
runs                     1.2.2
s3transfer               0.10.4
scikit-learn             1.5.2
scipy                    1.14.1
selinux                  3.3
sepolicy                 3.3
setools                  4.4.0
setuptools               70.1.0
six                      1.16.0
sniffio                  1.3.1
soupsieve                2.6
ssh-import-id            5.11
stack-data               0.6.3
starlette                0.22.0
starsessions             1.3.0
sympy                    1.12.1
tensorboard              2.18.0
tensorboard-data-server  0.7.2
threadpoolctl            3.5.0
torch                    2.0.0+cu118
torchaudio               2.0.1+cu118
torchmetrics             1.6.0
torchsummary             1.5.1
torchvision              0.15.1+cu118
tornado                  6.4.2
tqdm                     4.67.1
traitlets                5.14.3
triton                   2.0.0
types-python-dateutil    2.9.0.20241003
typing_extensions        4.12.2
tzdata                   2024.2
urllib3                  2.2.2
uvicorn                  0.32.1
wcwidth                  0.2.13
websocket-client         1.8.0
websockets               11.0.3
Werkzeug                 3.1.3
wheel                    0.43.0
xmod                     1.8.1
yarl                     1.18.3


in this env,you should change the config.yaml 
trainer_params:
  accelerator: 'gpu'
  devices: [0]
  max_epochs: 100
and     Comment out DDPS in run.py             
#strategy=DDPStrategy(find_unused_parameters=False),

**Comment by RobertoGrabovac** [comment]
Thank you so much, it seems like its working but training process is taking too long unfortunately. Could you maybe provide me with the last checkpoint of your model (both decoder and encoder), so that I can load it? I would like to be able to encode some of my pictures and decode them, to see how does the reconstructed pictures look. This would be very helpful to me and I thank you in advance. Sorry for taking your time, but you already helped me a lot :)

**Comment by ChenMuDX** [comment]
[Uploading vaeckp.zip…]()
I think what you want is in the zip file.

**Comment by ChenMuDX** [comment]
If the file can't be uploaded, give me your e-mail address and I'll send it to you.

**Comment by yanzilhu** [comment]
I am now reproducing VanillaVAE and training for 100 epochs, but the results are very poor. Could you please provide your weight file for reference?

**Comment by yanzilhu** [comment]
my gmail is yanzilh8@gmail.com thank you so much
